src/base/AnyMap.cpp

```diff
@@ -641,6 +641,66 @@ bool AnyValue::isScalar() const {
     return is<double>() || is<long int>() || is<std::string>() || is<bool>();
 }
 
+size_t AnyValue::vectorSize() const {
+    if (isVector<double>()) {
+        return as<vector<double>>().size();
+    }
+    if (isVector<long int>()) {
+        return as<vector<long int>>().size();
+    }
+    if (isVector<string>()) {
+        return as<vector<string>>().size();
+    }
+    if (isVector<bool>()) {
+        return as<vector<bool>>().size();
+    }
+    return npos;
+}
+
+pair<size_t, size_t> AnyValue::matrixShape() const {
+    if (isVector<vector<double>>()) {
+        auto& mat = as<vector<vector<double>>>();
+        if (isMatrix<double>()) {
+            if (mat.size()) {
+                return pair<size_t, size_t>(mat.size(), mat[0].size());
```

Comment:
I think here and in all following instances, you can just write:
```suggestion
                return {mat.size(), mat[0].size()};
```